x/pairing/keeper/scores/score_req.go

```diff
@@ -0,0 +1,159 @@
+package scores
+
+import (
+	"fmt"
+	"math/big"
+
+	sdk "github.com/cosmos/cosmos-sdk/types"
+	sdkerrors "github.com/cosmos/cosmos-sdk/types/errors"
+	"github.com/lavanet/lava/utils"
+	epochstoragetypes "github.com/lavanet/lava/x/epochstorage/types"
+	scorestypes "github.com/lavanet/lava/x/pairing/types/scores"
+	planstypes "github.com/lavanet/lava/x/plans/types"
+	tendermintcrypto "github.com/tendermint/tendermint/crypto"
+)
+
+var uniformStrategy scorestypes.ScoreStrategy
+
+// TODO: currently we'll use weight=1 for all reqs. In the future, we'll get it from policy
+func init() {
+	uniformStrategy = make(scorestypes.ScoreStrategy)
+	uniformStrategy[scorestypes.STAKE_REQ_NAME] = 1
+}
+
+// get the overall requirements from the policy and assign slots that'll fulfil them
+// TODO: this function should be changed in the future since it only supports stake reqs
+func CalcSlots(policy planstypes.Policy) []*scorestypes.PairingSlot {
+	slots := make([]*scorestypes.PairingSlot, policy.MaxProvidersToPair)
+	stakeReq := scorestypes.StakeReq{}
+	stakeReqMap := map[string]scorestypes.ScoreReq{stakeReq.GetName(): stakeReq}
+	for i := range slots {
+		slots[i] = scorestypes.NewPairingSlot(stakeReqMap)
+	}
+
+	return slots
+}
+
+// group the slots
+func GroupSlots(slots []*scorestypes.PairingSlot) []scorestypes.PairingSlotGroup {
+	slotGroups := []scorestypes.PairingSlotGroup{}
+	if len(slots) == 0 {
+		utils.LavaFormatError("no slots", sdkerrors.ErrLogic)
+		return slotGroups
+	}
+
+	for k := range slots {
+		foundGroup := false
+		for i := range slotGroups {
+			diff := slots[k].Diff(slotGroups[i].Slot)
+			if len(diff.Reqs) == 0 {
+				slotGroups[i].Count += 1
+				foundGroup = true
+				break
+			}
+		}
+
+		if !foundGroup {
+			newGroup := scorestypes.NewPairingSlotGroup(slots[k])
+			slotGroups = append(slotGroups, *newGroup)
+		}
+	}
+
+	return slotGroups
+}
+
+// TODO: currently we'll use weight=1 for all reqs. In the future, we'll get it from policy
+func GetStrategy() scorestypes.ScoreStrategy {
+	return uniformStrategy
+}
+
+// calculates the final pairing score for all slot groups (with strategy)
+// we calculate only the diff between the current and previous slot groups
+func CalcPairingScore(scores []*scorestypes.PairingScore, strategy scorestypes.ScoreStrategy, diffSlot *scorestypes.PairingSlot, minStake sdk.Int) error {
+	// calculate the score for each req for each provider
+	for _, req := range diffSlot.Reqs {
+		reqName := req.GetName()
+		weight, ok := strategy[reqName]
+		if !ok {
+			return utils.LavaFormatError("req not in strategy", sdkerrors.ErrKeyNotFound,
+				utils.Attribute{Key: "req_bitmap_value", Value: reqName})
+		}
+
+		for _, score := range scores {
+			// normalize stake so we won't overflow the score result (uint64)
+			providerWithNormalizedStake := score.Provider
```

Comment:
providerWithNormalizedStake is a pointer, meaning if we have mote than 1 req it will normalize it again, should be done in the stakescore.score

Reply:
done. see https://github.com/lavanet/lava/commit/b861439bb85b2ef959e9e3594ed7ed25f0946bc3